src/test/java/org/apache/mesos/scheduler/plan/DefaultPlanCoordinatorTest.java

```diff
@@ -0,0 +1,296 @@
+    SchedulerDriver schedulerDriver;
+
+    @Before
+    public void setupTest() {
+        MockitoAnnotations.initMocks(this);
+        offerAccepter = spy(new OfferAccepter(Arrays.asList()));
+        stateStore = mock(StateStore.class);
+        taskFailureListener = mock(TaskFailureListener.class);
+        schedulerDriver = mock(SchedulerDriver.class);
+        taskKiller = new DefaultTaskKiller(stateStore, taskFailureListener, schedulerDriver);
+        planScheduler = new DefaultPlanScheduler(offerAccepter, new OfferEvaluator(stateStore), taskKiller);
+        taskSets = Arrays.asList(TestTaskSetFactory.getTaskSet());
+        taskSetsB = Arrays.asList(TestTaskSetFactory.getTaskSet(
+                TestTaskSetFactory.NAME + "-B",
+                TestTaskSetFactory.COUNT,
+                TestTaskSetFactory.CMD.getValue(),
+                TestTaskSetFactory.CPU,
+                TestTaskSetFactory.MEM,
+                TestTaskSetFactory.DISK));
+        serviceSpecification = new DefaultServiceSpecification(
+                SERVICE_NAME,
+                taskSets);
+        serviceSpecificationB = new DefaultServiceSpecification(
+                SERVICE_NAME + "-B",
+                taskSetsB);
+    }
+
+    private List<Protos.Offer> getOffers(double cpus, double mem, double disk) {
+        final ArrayList<Protos.Offer> offers = new ArrayList<>();
+        offers.addAll(OfferTestUtils.getOffers(
+                Arrays.asList(
+                        ResourceTestUtils.getUnreservedCpu(cpus),
+                        ResourceTestUtils.getUnreservedMem(mem),
+                        ResourceTestUtils.getUnreservedDisk(disk))));
+        offers.add(Protos.Offer.newBuilder(OfferTestUtils.getOffers(
+                Arrays.asList(
+                        ResourceTestUtils.getUnreservedCpu(cpus),
+                        ResourceTestUtils.getUnreservedMem(mem),
+                        ResourceTestUtils.getUnreservedDisk(disk))).get(0))
+                .setId(Protos.OfferID.newBuilder().setValue("other-offer"))
+                .build());
+        return offers;
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNoPlanManager() {
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(Arrays.asList(), planScheduler);
+        coordinator.processOffers(schedulerDriver, getOffers(1, 1, 1));
+    }
+
+    @Test
+    public void testOnePlanManagerPendingSufficientOffer() throws Exception {
+        when(stateStore.fetchTask(anyString())).thenReturn(Optional.empty());
+        final Plan plan = new DefaultPlanFactory(stateStore).getPlan(serviceSpecification);
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(
+                Arrays.asList(new DefaultPlanManager(plan, new DefaultStrategyFactory())),
+                planScheduler);
+        Assert.assertTrue(coordinator.processOffers(schedulerDriver, getOffers(2, 2000, 10000)).size() == 1);
```

Comment:
Can we rename these constants to stuff like `SUFFICENT_CPU` and `INSUFFICIENT_MEM`.